client/web/web.go

```diff
@@ -505,6 +505,22 @@ func (s *Server) serveAPI(w http.ResponseWriter, r *http.Request) {
 			http.Error(w, "method not allowed", http.StatusMethodNotAllowed)
 		}
 		return
+	case path == "/update":
+		switch r.Method {
+		case httpm.POST:
+			s.serveSelfUpdate(w, r)
+		default:
+			http.Error(w, "method not allowed", http.StatusMethodNotAllowed)
+		}
+		return
+	case path == "/update/progress":
+		switch r.Method {
+		case httpm.GET:
+			s.serveUpdateProgress(w, r)
+		default:
+			http.Error(w, "method not allowed", http.StatusMethodNotAllowed)
+		}
+		return
```

Comment:
We could probably actually use the localapi proxy for this instead, since these should probably only be needed in the full, non-readonly, client. If you look at the localapi proxy lower in this file, you'll see an allowlist of endpoints that can be hit once the user has a valid session. If you use that, we don't need these extra handlers or the localclient functions.